### ISSUE TITLE
Typing certain emoji crashes the app

### ISSUE THREAD

**Issue by borisyankov** [zulip/zulip-mobile#2787]
This happens if using `Gboard - the Google Keyboard`.
This is reported as being the fix:
https://github.com/facebook/react-native/commit/41975f75d96ef4b606b4618461bf24d5db063b77 

The commit is included in React Native 0.56

**Comment by gnprice** [comment]
Sweet -- glad this is already fixed upstream!

Looks like https://github.com/facebook/react-native/issues/18405#issuecomment-373624413 is the RN issue report. (Followed a link from the commit message on the commit mentioned above; and the repro steps look like a perfect match.)

Though... hmm. That issue is still open.

... OK, but the linked commit (which came from https://github.com/facebook/react-native/pull/18443) is supposed to be a workaround that applies if you don't pass an `onKeyPress` prop to your `TextInput`. Which we currently don't, so I guess when we upgrade to RN 0.56 that will fix the issue for us.

**Comment by gnprice** [comment]
BTW that upstream issue thread helps clarify what the conditions are that trigger the issue: it looks like it's not specific to Gboard, but it happens on Android when you use any keyboard to type... actually any character outside [the Unicode BMP](https://en.wikipedia.org/wiki/Plane_(Unicode)#Basic_Multilingual_Plane), i.e. a character that doesn't fit into one Java `char`. In practice that mostly means an emoji that isn't one of the oldest emoji.

**Comment by gnprice** [comment]
Unassigning because I believe there isn't any ongoing work to do on this issue; it should just get fixed when we upgrade RN. (Though part of doing that upgrade is we should confirm that.) Thanks again @borisyankov for writing up the issue and for the research!

I just filed #2788 to track the RN upgrade. If you want to take that, please do :)

**Comment by gnprice** [comment]
Also: thanks @adnrs96 for originally reporting this issue!

**Comment by gnprice** [comment]
... Hmm, on the one hand I should have reread the thread more closely:

> BTW that upstream issue thread helps clarify what the conditions are that trigger the issue: it looks like it's not specific to Gboard, but it happens on Android when you use any keyboard to type... actually any character outside [the Unicode BMP](https://en.wikipedia.org/wiki/Plane_(Unicode)#Basic_Multilingual_Plane), i.e. a character that doesn't fit into one Java `char`. In practice that mostly means an emoji that isn't one of the oldest emoji.

So this is something any of us can easily reproduce with an Android emulator or device. :confetti_ball: (🎊 U+1F38a CONFETTI BALL) is one handy example of an emoji that should trigger the issue.

On the other hand, I just tried that on 19.2.102 (the prod release from yesterday, for Android) on my phone, and it didn't reproduce! So there's something here we don't understand.

Maybe Android 9 fixed this? My phone is on that now; it was Android 8 (maybe 8.1) when we looked at this last, in July.

This is probably still resolved by that RN upgrade (#2789), but it'd be good to poke around at least a little more to confirm we still know some way to reproduce the issue before that putative fix. We're not aware of anything in 19.2.192, relative to a few months ago when this was reported, that would have fixed it.
